Sources/Synchronization-swift/SynchronizeOperation.swift

```diff
@@ -0,0 +1,57 @@
+//
+//  SynchronizeOperation.swift
+//  Synchronization-swift
+//
+//  Created by Evegeny Kalashnikov on 07.08.2020.
+//
+
+import Foundation
+
+public protocol AnySynchronizeOperationListener: class {
+    func statusDidUpdated(status: SyncStatus, model: AnySynchronizable, rootModel: AnySynchronizable)
+    func didSynced(node: SyncNode, id: Int)
+    func didFailed(node: SyncNode, error: Error)
+}
+
+public class SynchronizeOperation: Operation, AnySynchronizationStatusListener {
+
+    private let node: SyncNode
+    private let manager: AnySynchronizationManager
+    private weak var listener: AnySynchronizeOperationListener?
+    private weak var customPerformer: AnySynchronizationCustomPerformer?
+
+    public func isEqual(to iddentifier: SyncId) -> Bool {
+        return self.node.value.syncId == iddentifier
+    }
+
+    public func isEqual(to node: SyncNode) -> Bool {
+        return self.node == node
+    }
+
+    public init(node: SyncNode,
+                manager: AnySynchronizationManager,
+                listener: AnySynchronizeOperationListener,
+                customPerformer: AnySynchronizationCustomPerformer? = nil) {
+
+        self.node = node
+        self.manager = manager
+        self.listener = listener
+        self.customPerformer = customPerformer
+    }
+
+    public override func main() {
+        super.main()
+        do {
+            let rootId = try self.manager.synchronize(node: self.node, listener: self, customPerformer: self.customPerformer)
+                .wait()
```

Comment:
what do you think about the next formatting style, since synchronize and wait are serial operations:

```
let rootId = try self.manager
                .synchronize(node: self.node, listener: self, customPerformer: self.customPerformer)
                .wait()
```

?